### PR TITLE
Add support for global pooling for trt

### LINKED ISSUE #12754: support global_pooling in tensorrt op converter
support global_pooling in tensorrt op converter.

### PR DESCRIPTION
**Pull request by NHZlX**
fix #12754

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/convert/test_pool2d_op.cc [hunk 1]
```diff
@@ -40,11 +40,13 @@ TEST(Pool2dOpConverter, main) {
   std::vector<int> strides({2, 2});
   std::vector<int> paddings({0, 0});
   std::string pooling_t = "max";
+  bool global_pooling = false;
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/test_pool2d_op.cc]: any need to test both true and false?

Comment (NHZlX) [reply paddle/fluid/inference/tensorrt/convert/test_pool2d_op.cc]: we don't need the care the test attr, we always in the test mode.

Comment (NHZlX) [reply paddle/fluid/inference/tensorrt/convert/test_pool2d_op.cc]: The test for global_pooling (true) is needed.

Comment (NHZlX) [reply paddle/fluid/inference/tensorrt/convert/test_pool2d_op.cc]: Done!

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**